src/metrics/counter.ts

```diff
@@ -10,8 +12,10 @@ export function makeCounterProvider(
 ): Provider {
   return {
     provide: getToken(options.name),
-    useFactory(): client.Metric<string> {
+    useFactory(config: PrometheusPrefix): client.Metric<string> {
+      options.name = config.prefix.concat(options.name);
       return getOrCreateMetric("Counter", options);
```

Comment:
I would rather you update the signature of `getOrCreateMetric` to take an optional last arg of the options:

```ts
getOrCreateMetric(name: string, options, prometheusOptions?: PrometheusOptions)
```

And then add the prefixing logic there.

Reply:
Make it optional to preserve backwards compatibility